clippy_lints/src/default.rs

```diff
@@ -94,6 +96,7 @@ impl<'tcx> LateLintPass<'tcx> for Default {
             if let QPath::Resolved(None, _path) = qpath;
             let expr_ty = cx.typeck_results().expr_ty(expr);
             if let ty::Adt(def, ..) = expr_ty.kind();
+            if !is_from_proc_macro(cx, expr);
```

Comment:
I would say there aren't many lints which you would want to. Both correctness and perf lints would be the only candidates and even then it could be more noisy than would it would be worth.

Reply:
This is definitely a separate discussion from this PR though.